### ISSUE TITLE
Generics

### ISSUE THREAD

**Issue by realdadfish** [beworker/magnet#80]
So I guess this came up before, but anyways :) I want to be able to inject generic implementations and right now Magnet hinders me to do that because of some class name validation issue:

```kotlin
const val RATINGS_STORAGE = "ratings-storage"

data class RatingsModel(...)

interface ModelStore<T> {
    fun load(): T?
    fun save(data: T)
}

@Instance(type = ModelStore::class, classifier = RATINGS_STORAGE)
fun provideRatingsModelStore(): ModelStore<RatingsModel> {
    return /* create the store for the RatingsModel */ 
    }
}

@Instance(type = RatingsManager::class)
class RatingsManager internal constructor(
    @Classifier(RATINGS_STORAGE) private val store: ModelStore<RatingsData>
) {
   ...
}
```

This fails with

```
error: Method must return instance of path.to.ModelStore as declared by interface magnet.Instance
    public static final de.aoksystems.ma.abp.modelstore.ModelStore<RatingsData> provideRatingsModelStore()
```

Now I thought maybe I could outsmart the validation and use 

```kotlin
fun provideRatingsModelStore(): ModelStore<*> { ... }
```

but this didn't work either. And naturally, Kotlin doesn't let me use the plain `ModelStore` type without any generic arguments, as I could do with Java:

```java
public class StaticProvision {
    @Instance(type = ModelStore.class, classifier = RATINGS_STORAGE)
    static ModelStore provideRatingsModelStore() {
        return /* create the store for the RatingsModel */;
    }
}
```

This version of course compiles just fine.

What are my options here (beside starting to write Java code again :))?

**Comment by realdadfish** [comment]
That was quick, thank you! 🙏

**Comment by beworker** [comment]
You are welcome. The fix is available in `3.1-beta1`.
